### ISSUE TITLE
Incorporate newer `key` API method in place of the deprecated `which` method

### ISSUE THREAD

**Issue by rexagod** [publiclab/Leaflet.DistortableImage#211]
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/Leaflet.DistortableImage/labels/help%20wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update:** In order to remove support for the deprecated `which` method, we'll be replacing it by the `key` method, while making some other appropriate changes as well. The steps are as follows.

- Replace the following key map values to comply with the `key` method, i.e., with the keymap provided below it.
https://github.com/publiclab/Leaflet.DistortableImage/blob/b84229e727538e44f7e386243161b9bae379dd41/src/edit/DistortableImage.Edit.js#L7-L22
```js
		keymap: {
			'Backspace': '_removeOverlay',
			'Delete': '_removeOverlay',
			'CapsLock': '_toggleRotate',
			'Escape': '_deselect',
			'd': '_toggleRotateDistort',
			'r': '_toggleRotateDistort',
			'i': '_toggleIsolate',
			'j': '_sendUp',
			'k': '_sendDown',
			'l': '_toggleLock',
			'o': '_toggleOutline',
			's': '_toggleScale',
			't': '_toggleTransparency',
		}
```
- Let's replace the `which` method with the `key` one as follows.
https://github.com/publiclab/Leaflet.DistortableImage/blob/b84229e727538e44f7e386243161b9bae379dd41/src/edit/DistortableImage.Edit.js#L203
```js
handlerName = keymap[event.key];
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

**Comment by rexagod** [comment]
Pinging @jywarren, @sashadev-sky, or @gauravano to mark this as an `fto`. Please refer #178 for context.

**Comment by sashadev-sky** [comment]
@rexagod ok cool! Did you look up if this is the new convention for keymaps? I know I said `key` but like I said wasn't positive!! But it makes sense, right?

**Comment by sashadev-sky** [comment]
making a candidate until you get back to me on that. Also did you make these changes locally and everything still worked?

**Comment by rexagod** [comment]
@sashadev-sky I did look up the convention and [compatibility](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#Browser_compatibility) of the `key` method and this seems to fit in. Also, I've verified this locally, you can be sure.

**Comment by sashadev-sky** [comment]
@rexagod Looks good! Changing this to an FTO. I'll post it in the plots2 FTO page for you so people see it. Is the plan to update all of them through a bunch of FTOs?

**Comment by SteliosPhanartzis** [comment]
Can I claim this?

**Comment by sashadev-sky** [comment]
@SteliosPhanartzis Yes all yours!

**Comment by SteliosPhanartzis** [comment]
Was the DistortableImage.Edit.js the only file that had the which method?

**Comment by sashadev-sky** [comment]
@SteliosPhanartzis yes the only one that's currently in use. Don't worry about standardizing with the rest of the repo, the FTOs are meant to be a smaller, step by step situation. 

You can open a new PR once you're done with this to make more updates if you'd like!

**Comment by sashadev-sky** [comment]
@rexagod I am just going to resolve this one in #229 was a good FTO but something that should just be fixed in the rest of my cleanup at this point. I will open some more FTOs in this repo shortly though!

**Comment by sashadev-sky** [comment]
@rexagod I think that `_removeOverlay` will now just be 'Backspace' because the <kbd>delete</kbd> key on mac and the <kbd>backspace</kbd> on windows actually both trigger `event.key` 'Backspace'.

I'm thinking this is sufficient, and we don't need to add the 2nd way. I originally thought it "evened them out" but now I just think its unnecessary. Do you think it's ok to proceed with just `Backscape` and write a comment next to it: " backspace windows / delete mac "

**Comment by rexagod** [comment]
Sure, we can definitely proceed with this and adding an appropriate comment at the same time should suffice the user's understanding of this feature. :+1:

**Comment by sashadev-sky** [comment]
Ok cool, you're already a reviewer on my PR but i'll ping you again there for final review soon @rexagod

**Comment by sashadev-sky** [comment]
closed via #229
